rmf_task/include/rmf_task/requests/ParkRobotFactory.hpp

```diff
@@ -44,8 +44,13 @@ class ParkRobotFactory : public RequestFactory
   ///   If nullopt, the AGV will return to its charging_waypoint and remain idle
   ///   there. It will not wait for its battery to charge up before undertaking
   ///   new tasks.
+  ///
+  /// \param[in] requester
+  ///   The identifier for the entity that would make parking requests. This
+  ///   field is optional.
   ParkRobotFactory(
-    std::optional<std::size_t> parking_waypoint = std::nullopt);
+    std::optional<std::size_t> parking_waypoint = std::nullopt,
+    std::optional<std::string> requester = std::nullopt);
```

Comment:
Move to `make_request()` as suggested above. Ideally even the `parking_waypoint` is passed to `make_request()` but I think we went this constructor as we don't have a reservation system in place.